components/camel-rest-swagger/src/main/java/org/apache/camel/component/rest/swagger/RestSwaggerEndpoint.java

```diff
@@ -308,14 +308,24 @@ RestSwaggerComponent component() {
         return (RestSwaggerComponent) getComponent();
     }
 
+    @Override
```

Comment:
I wonder why sonarcloud reports a problem that equals/hashCode needs to be overridden when they just call super?

Reply:
Sonar cares about additional [subclass fields and the needeed of add them in equals and hash code methods](https://rules.sonarsource.com/java/RSPEC-2160). In that case this is not necessary as equality is based on context name and endpoint Uri so additional fields dont need to be considered. However sonar points as issue.